comp/networkpath/npcollector/npcollectorimpl/npcollector.go

```diff
@@ -193,8 +193,9 @@ func (s *npCollectorImpl) runTracerouteForPath(ptest *pathteststore.PathtestCont
 	cfg := traceroute.Config{
 		DestHostname: ptest.Pathtest.Hostname,
 		DestPort:     ptest.Pathtest.Port,
-		MaxTTL:       0, // TODO: make it configurable, setting 0 to use default value for now
-		TimeoutMs:    0, // TODO: make it configurable, setting 0 to use default value for now
+		MaxTTL:       0,              // TODO: make it configurable, setting 0 to use default value for now
+		TimeoutMs:    0,              // TODO: make it configurable, setting 0 to use default value for now
+		Protocol:     traceroute.TCP, // TODO: would we ever want UDP here?
```

Comment:
I think we should use the protocol from the NPM conns, some changes are needed to pass protocol from `npCollectorImpl.ScheduleConns()` to `PathtestContext` and then to `traceroute.Config` here.

Reply:
Created a separate PR to merge into this one that handles this: https://github.com/DataDog/datadog-agent/pull/26110